backend-rust/src/indexer.rs

```diff
@@ -415,6 +417,24 @@ LIMIT 1
         .await
         .context("Failed to query data for save context")?;
 
+        let last_block = sqlx::query!(
+            r#"
+SELECT
+  slot_time
+FROM blocks
+ORDER BY height DESC
+LIMIT 1
```

Comment:
What if an insertion/update happens into the blocks table between this query and the previous one? Is that okay? If not, maybe a transaction with a lock would make sense.

Reply:
This would only happen if running multiple indexers, which we don't support for now.
At some later point, we should warn the user and prevent this from happening.